src/resources/customers/index.ts

```diff
@@ -0,0 +1,218 @@
+ * The `customers` resource
+ */
+export default class CustomersResource extends CustomersBaseResource {
+  public static resource = 'customers';
+  public static model = Customer;
+  public apiName = 'Customers API';
+
+  /**
+   * List Customers.
+   *
+   * @since 2.0.0
+   *
+   * @see https://docs.mollie.com/reference/v2/customers-api/list-customers
+   * @public ✓ This method is part of the public API
+   * @alias list
+   */
+  public all = this.list;
+  /**
+   * List Customers.
+   *
+   * @since 2.2.0
+   *
+   * @see https://docs.mollie.com/reference/v2/customers-api/list-customers
+   * @public ✓ This method is part of the public API
+   * @alias list
+   */
+  public page = this.list;
+  /**
+   * Delete a Customer.
+   *
+   * @since 2.0.0
+   *
+   * @see https://docs.mollie.com/reference/v2/customers-api/delete-customer
+   * @public ✓ This method is part of the public API
+   * @alias delete
+   */
+  public cancel = this.delete;
+
+  /**
+   * Creates a simple minimal representation of a customer in the Mollie API
+   * to use for the {@link https://www.mollie.com/en/checkout Mollie Checkout}
+   * and Recurring features.
+   * These customers will appear in your
+   * {@link https://www.mollie.com/dashboard/ Mollie Dashboard}
+   * where you can manage their details,
+   * and also see their payments and subscriptions.
+   *
+   * @param params - Create customer parameters
+   * @param cb - (DEPRECATED SINCE 2.2.0) Callback function, can be used instead of the returned `Promise` object
+   *
+   * @returns The newly created customer object
+   *
+   * @since 2.0.0
+   *
+   * @see https://docs.mollie.com/reference/v2/customers-api/create-customer
+   * @public ✓ This method is part of the public API
+   */
+  public async create(params: ICreateParams, cb?: CreateCallback): Promise<Customer> {
+    return super.create(params, cb) as Promise<Customer>;
+  }
+
+  /**
+   * Retrieve a single customer by its ID
+   *
+   * @param id - Customer ID
+   * @param params - Retrieve customer parameters
+   *                 (DEPRECATED SINCE 2.2.0) Can also be a callback function
+   * @param cb - (DEPRECATED SINCE 2.2.0) Callback function, can be used instead of the returned `Promise` object
+   *
+   * @returns Customer object
+   *
+   * @since 2.0.0
+   *
+   * @see https://docs.mollie.com/reference/v2/customers-api/get-customer
+   * @public ✓ This method is part of the public API
+   */
+  public async get(id: string, params?: IGetParams | GetCallback, cb?: GetCallback): Promise<Customer> {
+    // Using callbacks (DEPRECATED SINCE 2.2.0)
+    if (typeof params === 'function' || typeof cb === 'function') {
+      if (!startsWith(id, Customer.resourcePrefix)) {
+        throw Resource.errorHandler(
+          { error: { message: 'The customer id is invalid' } },
+          typeof params === 'function' ? params : cb,
+        );
+      }
+
+      return super.get(
+        id,
+        typeof params === 'function' ? null : params,
```

Comment:
We can abstract this into a separate function (`getParameters` or something similar) and remove this duplication from other resources as well.

Reply:
`useParameters()` ^^

Reply:
I think it might be a good idea to start abstracting this when we can remove the deprecated parts. This call to `super.get` will have to be changed with the next major version anyway.

Reply:
Hmm, not sure to be honest. I would like to prevent introducing breaking changes as long as possible, especially since we already introduced breaking changes a few months ago. I'm really happy that the typescript migration allows us to stay backward compatible.

Reply:
Yeah, we should only add new functionality. I meant postponing these kinds of abstractions until v3 or any other major version. Best to keep it as is for now.